include/swift/AST/DiagnosticsSema.def

```diff
@@ -4985,8 +4985,8 @@ NOTE(candidate_is_not_assignable,none,
      (const ValueDecl *))
 
 NOTE(change_to_mutating,none,
-     "mark %select{method|accessor}0 'mutating' to make 'self' mutable",
-     (bool))
+     "mark %select{method|%1}0 'mutating' to make 'self' mutable",
+     (bool, StringRef))
```

Comment:
Why pass a `StringRef` and not a `DescriptiveDeclKind`? Or better:

```suggestion
     "mark %select{method|%kindonly1}0 'mutating' to make 'self' mutable",
     (bool, FuncDecl *))
```

...and pass the function decl itself.